prometheus/src/main/java/org/opensearch/sql/prometheus/planner/logical/PrometheusLogicalPlanOptimizerFactory.java

```diff
@@ -23,10 +28,10 @@ public class PrometheusLogicalPlanOptimizerFactory {
    * Create Prometheus storage specified logical plan optimizer.
    */
   public static LogicalPlanOptimizer create() {
-    return new LogicalPlanOptimizer(Arrays.asList(
+    return new LogicalPlanOptimizer(Stream.of(
         new MergeFilterAndRelation(),
         new MergeAggAndIndexScan(),
         new MergeAggAndRelation()
-    ));
+    ).map(r -> (Rule<LogicalPlan>)r).collect(Collectors.toList()));
```

Comment:
this is messy, and we can update the Rules classes to extend `LogicalPlan` instead.